DeutschJozsaAlgorithm/Tasks.qs

```diff
@@ -162,14 +162,58 @@ namespace Quantum.Kata.DeutschJozsaAlgorithm {
         // Hint: represent f(x) in terms of AND and ⊕ operations
 
         // ...
+    } 
+    
+    //////////////////////////////////////////////////////////////////
+    // Part II. Deutsch-Jozsa Algorithm
+    //////////////////////////////////////////////////////////////////
+    
+    // Task 2.1. Deutsch-Jozsa algorithm implementation
+    // Inputs:
+    //      1) the number of qubits in the input register N for the function f
+    //      2) a quantum operation which implements the oracle |x⟩|y⟩ -> |x⟩|y ⊕ f(x)⟩, where
+    //         x is an N-qubit input register, y is a 1-qubit answer register, and f is a Boolean function
+    // You are guaranteed that the function f implemented by the oracle is either
+    // constant (returns 0 on all inputs or 1 on all inputs) or
+    // balanced (returns 0 on exactly one half of the input domain and 1 on the other half).
+    // Output:
+    //      true if the function f is constant
+    //      false if the function f is balanced
+    //
+    // Note: a trivial approach is to call the oracle multiple times:
+    //       if the values for more than half of the possible inputs are the same, the function is constant.
+    // Quantum computing allows to perform this task in just one call to the oracle; try to implement this algorithm.
+    operation DJ_Algorithm (N : Int, Uf : ((Qubit[], Qubit) => Unit)) : Bool {
+        
+        // Declare Bool variable in which the result will be accumulated;
+        // this variable has to be mutable to allow updating it.
+        mutable isConstantFunction = true;
+        
+        // ...
+
+        return isConstantFunction;
     }
     
     
+    // Task 2.2. Testing Deutsch-Jozsa algorithm
+    // Goal: use your implementation of Deutsch-Jozsa algorithm from task 3.1 to test
+    // each of the oracles you've implemented in part I for being constant or balanced.
+    operation DJ_Test () : Unit {
+        // Hint: you will need to use partial application to test oracles such as Oracle_Kth_Qubit and Oracle_ProductFunction;
```

Comment:
We're going to need to swap hints from testing tasks on DJ and BV algorithms - the detailed hint needs to go into the first task the learner sees, and the second one needs to refer to the first one.